### ISSUE TITLE
[NEW] Support multiples servers 

### ISSUE THREAD

**Issue by rafaelks** [RocketChat/Rocket.Chat.Android#839]
## General Information

The application needs to support multiple servers to be authenticated and let the user change it from the interface (described below).

### How to persist information

The information for now can be saved in a specie of "keychain" in Android, since we don't have any database model yet. The information needs to be securely encrypted.

### Interface

For now, let's simply add new items to the top of the left-sidebar will show the currently selected item and an option to show all available accounts (image below). When pressing it, would a new segment to show all the list of available servers and an option to add a new one.

![IMG_0025.jpg](https://images.zenhubusercontent.com/5996eb0e8f62dc7798c34549/a93fdf16-c3b0-4dd4-b595-a14ae2dfd6d7)

**Comment by PVince81** [comment]
Hmm there's something I don't understand: this PR https://github.com/RocketChat/Rocket.Chat.Android/pull/436 seems to add support for multiple servers somehow but this ticket here seems to imply that the feature isn't there yet.

And there's also https://github.com/RocketChat/Rocket.Chat.Android/issues/567.

How much of this is already implemented ?

**Comment by rafaelks** [comment]
@PVince81 You're right, this is confusing. Let me try to explain:

1. We do have multi server support in our version 1.
2. This issue is addressed to milestone of version 2, it's a rewritten application, so we'll need to implement it twice;
3. The issue #567 is a more robust support; We don't have all the details to share for now, but is something like this: the app could request an URL and it would contain all the servers authenticated in the app, so the user would only need to authenticate once to have (for example) 10 servers authenticated in the mobile app; Did I made myself clear?

Hopefully I could make it more clear to you. Let me know if you have any other question please.

**Comment by PVince81** [comment]
@rafaelks thanks for clarification. I didn't know there was a rewrite planned! (it's also the first time I visit this project's bug tracker)

In the end I found the multi server dialog in version 1 with the double finger swipe up.

For the authentication part would version 2 integrate with Android's accounts support ? (the part that can be seen in the settings where all other accounts are). As far as I saw even version 1 is requesting the matching permission for this but doesn't seem to be actually using it. Maybe that's the keychain part you mentioned.
